ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/requesthandler/internalresponse/EthSignBodyProvider.java

```diff
@@ -0,0 +1,117 @@
+
+public class EthSignBodyProvider implements BodyProvider {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final TransactionSignerProvider transactionSignerProvider;
+
+  public EthSignBodyProvider(final TransactionSignerProvider transactionSignerProvider) {
+    this.transactionSignerProvider = transactionSignerProvider;
+  }
+
+  @Override
+  public JsonRpcBody getBody(final JsonRpcRequest request) {
+    try {
+      @SuppressWarnings("unchecked")
+      final List<String> params = (List<String>) request.getParams();
+      if (params == null || params.size() != 2) {
+        LOG.info(
+            "eth_sign should have a list of 2 parameters, but has {}",
+            params == null ? "null" : params.size());
+        return new JsonRpcBody(INVALID_PARAMS);
+      }
+      final String address = params.get(0);
+      final Optional<TransactionSigner> transactionSigner =
+          transactionSignerProvider.getSigner(address);
+      if (transactionSigner.isEmpty()) {
+        LOG.info("Address ({}) does not match any available account", address);
+        return new JsonRpcBody(SIGNING_FROM_IS_NOT_AN_UNLOCKED_ACCOUNT);
+      }
+      final TransactionSigner signer = transactionSigner.get();
+      final String originalMessage = params.get(1);
+      final String message =
+          (char) 25 + "Ethereum Signed Message:\n" + originalMessage.length() + originalMessage;
+      final Signature signature = signer.sign(message.getBytes(StandardCharsets.UTF_8));
+
+      final Bytes outputSignature =
+          Bytes.concatenate(
+              Bytes32.leftPad(Bytes.wrap(bigIntegerToBytes(signature.getR()))),
+              Bytes32.leftPad(Bytes.wrap(bigIntegerToBytes(signature.getS()))),
+              Bytes.wrap(bigIntegerToBytes(signature.getV())));
+      final JsonRpcSuccessResponse response =
+          new JsonRpcSuccessResponse(
+              request.getId(), Numeric.toHexString(outputSignature.toArray()));
+      return new JsonRpcBody(Json.encodeToBuffer(response));
+    } catch (final ClassCastException e) {
+      LOG.info(
+          "eth_sign should have a list of 2 parameters, but received an object: {}",
+          request.getParams());
+      return new JsonRpcBody(INVALID_PARAMS);
+    } catch (final Exception e) {
+      LOG.info("Unexpected error", e);
+      return new JsonRpcBody(INTERNAL_ERROR);
+    }
+  }
+
+  /**
+   * Omitting sign indication byte. <br>
+   * <br>
+   * Instead of {@link org.bouncycastle.util.BigIntegers#asUnsignedByteArray(BigInteger)} <br>
+   * we use this custom method to avoid an empty array in case of BigInteger.ZERO
+   *
+   * @param value - any big integer number. A <code>null</code>-value will return <code>null</code>
+   * @return A byte array without a leading zero byte if present in the signed encoding.
+   *     BigInteger.ZERO will return an array with length 1 and byte-value 0.
+   */
+  public static byte[] bigIntegerToBytes(final BigInteger value) {
```

Comment:
Does it make sense to move this method in a Util style class? And add few unit test case around it? Other than that it looks fine to me. @rain-on, @jframe FYI

Reply:
Cool, I moved it into `tech.pegasys.ethsigner.core.util.ByteUtils` and added some tests. Let me know if you prefer that I put it somewhere else